graphbolt/src/shared_memory_utils.h

```diff
@@ -20,52 +20,106 @@ namespace graphbolt {
 namespace sampling {
 
 /**
- * @brief SharedMemoryTensors includes: (1) two share memory objects holding
- * tensor meta information and data respectively; (2) a vector of optional
- * tensors on shared memory.
- */
-using SharedMemoryTensors = std::tuple<
-    SharedMemoryPtr, SharedMemoryPtr,
-    std::vector<torch::optional<torch::Tensor>>>;
-
-/**
- * @brief Copy torch tensors to shared memory.
- *
- * To simpilfy this interface, a regular tensor is also wrapped as an optional
- * one.
+ * @brief SharedMemoryHelper is a helper class to write/read data structures
+ * to/from shared memory.
  *
- * The function has two steps:
- * 1. Copy meta info to shared memory `shared_memory_name + "_meta"`. This is to
- * make sure that other loading processes can get the meta info of tensors.
- * 2. Copy tensors to shared memory `shared_memory_name + "_data"`, which can be
- * loaded by other processes with meta info.
+ * In order to write data structure to shared memory, we need to serialize the
+ * data structure to a binary buffer and then write the buffer to the shared
+ * memory. However, the size of the binary buffer is not known in advance. To
+ * solve this problem, we use two shared memory objects: one for storing the
+ * meta data and the other for storing the binary buffer. The meta data includes
```

Comment:
%s /meta data/metadata/g
```suggestion
 * metadata and the other for storing the binary buffer. The metadata includes
```